cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1092,9 +1110,24 @@ class SlurmComputeResourceSchema(_ComputeResourceSchema):
     disable_simultaneous_multithreading = fields.Bool(metadata={"update_policy": UpdatePolicy.COMPUTE_FLEET_STOP})
     schedulable_memory = fields.Int(metadata={"update_policy": UpdatePolicy.QUEUE_UPDATE_STRATEGY})
 
+    @validates_schema
+    def no_coexist_instance_type_flexibility(self, data, **kwargs):
+        """Validate that 'instance-type' and 'instance-types' do not co-exist."""
+        if self.fields_coexist(
+            data,
+            ["instance_type", "instance_type_list"],
+            one_required=True,
+            **kwargs,
+        ):
+            raise ValidationError(
+                "A Compute Resource needs to specify either a single InstanceType or a list of InstanceTypes."
+            )
+
     @post_load
     def make_resource(self, data, **kwargs):
         """Generate resource."""
+        if data.get("instance_type_list"):
+            return SlurmFlexibleComputeResource(**data)
         return SlurmComputeResource(**data)
```

Comment:
What about "A Compute Resource needs to specify either InstanceType or InstanceTypeList."